### ISSUE TITLE
Cannot see any of the previously sent messages anymore

### ISSUE THREAD

**Issue by lbriais** [wireapp/wire-desktop#1279]
WireInternal : Version Development
Wire Web Version 2018.02.01.1026
Operating system: Ubuntu 16.04

Regularly I get a message at the top of WireInternal, saying that a new version is available, I did update numerous time with no issue. But this time after I performed this update, __none of the messages sent by this device was visible anymore__.
All of them are replaced by an error message  (`211`) followed by the ID of the WireInternal device used to send the message originally.
I tried the "_reset session_" thing as stated in the error message with no success.

On other devices (an android app on my phone and another WireInternal but on another laptop not updated) the messages are still visible, but on the updated machine both WireInternal and the web UI in Firefox can't display messages sent by the WireInternal app anymore (I actually almost never use the web UI), but I guess the WireInternal app and the Web UI share the same code base...

This is really scary because if would only use this device, I would have loose all my messages, which would definitely be a show-stopper to use Wire anymore...

Please tell me there is a way to recover...

Thank you in advance

- Laurent -

**Comment by lbriais** [comment]
This is what I get in the UI for all past messages (Sorry it is in French)

![2018_02_15_selection_001](https://user-images.githubusercontent.com/3846669/36261110-966be7c0-1263-11e8-9920-94092ac40678.jpg)

**Comment by gregor** [comment]
@lbriais This is very unfortunate. Seems like the database was corrupted in one key point when you updated. We forgot the ID of the last event we decrypted and thuse tried to decrypt all messages you received over the past 28 days again. You should see your history in those chats if you scroll up far enough. Can you please check that?

That being said: We advise against using anything but the commits tagged as release for the individual platforms as your daily driver. The development build of the desktop client points at a pre-release of the webapp as well. This is beta quality and you can always run into issues.

**Comment by lbriais** [comment]
Agreed for the releases vs dev builds... Will definitely switch to the "official" binary releases you provide (but it also probably means that this "new" client will not see past messages...).

I scrolled up (for quite a while) and this is true that after a limit, messages are there normally, although I did not clearly see the 28 days limit. The question is now is there a way to recover ?

Thank you in advance for your support

**Comment by lbriais** [comment]
I had a look at all my past conversations, and actually there are 2 bizarre things:
* The date of the last correctly decrypted message does not _seem_ to be consistent across all conversations (sometimes difficult to say because I use 3 different devices)
* I found some messages __not sent by me__ that have also an error (`212` and not `211`), but in the middle of a conversation where some messages before and after the error are correctly decrypted...

Very disturbing...

**Comment by gregor** [comment]
Yes, you can recover:

1. Open the "Preferences", select "Devices" and make note of the device ID of the desktop app.
2. Close the desktop app.
3. Delete the device of the desktop appfrom another device by removing the device with the correct ID from the list.
4. Start the desktop app. You will be logged out.
5. Sign in again.

Your history will be there. The decryption errors will also be there. New messages will decrypt fine. New messages from you can be decrypted by others. You would remain on the dev builds. Writing a database query that removes all the decryption errors is possible, but this is definitely not supported.

The things you mentioned are also not bizarre. It depends on the state of the decryption session between two devices and the number of messages those clients exchanged. Especially whether the conversation was one-way only or both ways.

**Comment by lbriais** [comment]
Thank you for your support, but I'm not sure I completely got what you are asking me. I have at least few questions.

First by recover I meant a way to see again the messages that I see today with an error, and from your answer, what I understand is that the final result would be the same as my current status, ie:

* I will be able to see my messages older than a certain limit (which I currently already can)
* I will see everything that is in error now, still with errors (so no change)
* I will be able to send and receive messages from my friends correctly (which is already the case)

Tell me if I am wrong but if it is like that, I don't see any "_recovery_" there.

My questions are:

* We are talking there about moving away from the "dev" version to the "release" version, aren't we ? I mean are you telling me that if I install the `.deb` release I will see my history as it is currently within the "dev" version ? 
* In the step 3 of your procedure, do you mean "Delete the device of the desktop app from **any other** device" ? I guess you are talking about my 2 other devices, nothing to do with the people I am chatting with ?

Thank you very much for your time

**Comment by gregor** [comment]
Yes, you are right. The _recovery_ I was referring to will only ensure that future messages will not fail to decrypt as the state of your sessions could have been affected by the issue you hit.

To answer your questions:
1. No, when you move to the release version you will lose your history. There are unsupported, non advised ways to avoid that by copying the browser IndexedDB from the folder location of WireInternal to Wire.

2. Yes, the only location you can delete your devices from are your other two devices. No other people involved in this process.
